js/adapt-assessmentArticleModel.js

```diff
@@ -379,15 +375,67 @@ define([
             this.set("_isPass", isPass);
         },
 
-        _isAttemptsLeft: function() {
+        _getMarkingSettings: function() {
+            var markingSettings = {};
+
+            if (this._shouldSuppressMarking()) {
+                markingSettings = {
+                    _canShowMarking: false,
+                    _canShowModelAnswer: false,
+                    _canShowFeedback: false
+                };
+            } else {
+                var questionConfig = this.getConfig()._questions;
+                if (questionConfig.hasOwnProperty('_canShowFeedback')) {
+                    markingSettings._canShowFeedback = questionConfig._canShowFeedback;
+                }
+                
+                if (questionConfig.hasOwnProperty('_canShowModelAnswer')) {
+                    markingSettings._canShowModelAnswer = questionConfig._canShowModelAnswer;
+                }
+
+                if (questionConfig.hasOwnProperty('_canShowMarking')) {
+                    markingSettings._canShowMarking = questionConfig._canShowMarking;
+                }
+            }
+
+            return markingSettings;
+        },
+
+        _overrideMarkingSettings: function() {
+            var questionConfig = this.getConfig()._questions;
+            var newMarkingSettings = this._getMarkingSettings();
+            for (var i = 0, l = this._currentQuestionComponents.length; i < l; i++) {
+                this._currentQuestionComponents[i].set(newMarkingSettings, {
+                    pluginName: "_assessment"
+                });
+            }
+        },
+        _refreshQuestions: function() {
+            for (var a = 0, b = this._currentQuestionComponents.length; a < b; a++) {
+                var question = this._currentQuestionComponents[a];
+                question.refresh();
+            }
+        },
+        _shouldSuppressMarking: function() {
+            return this._isMarkingSuppressionEnabled() && this._isAttemptsEnabled() && this.get('_attemptsLeft') > 0;
+        },
```

Comment:
Should we be using the existing function `_isAttemptsLeft()` here rather than `this.get('_attemptsLeft') > 0`?